tools/autograd/gen_autograd_functions.py

```diff
@@ -462,6 +462,11 @@ def save_var(var: SavedAttribute, is_output: bool) -> None:
 
     if uses_single_grad(info):
         body.append('auto& grad = grads[0];')
+    else:
+        # Generate aliases for gradients named for returned values.
```

Comment:
interesting. one reason this would be better is to eliminate some potential confusion with second derivatives. The downside would be that it is a bit more verbose.

I don't have a sense for which is the bigger concern.